### PR TITLE
More timestamp tests and runtime functions

### PR DESCRIPTION
**Pull request by mbudiu-vmw**

### CODE REVIEW

File: SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/postgres/PostgresDateTests.java [hunk 1]
```diff
@@ -514,17 +1437,114 @@ public void testDoy() {
...
 
+    // SELECT EXTRACT(TIMEZONE      FROM DATE '2020-08-11');
+    //ERROR:  unit "timezone" not supported for type date
+    //SELECT EXTRACT(TIMEZONE_M    FROM DATE '2020-08-11');
+    //ERROR:  unit "timezone_m" not supported for type date
+    //SELECT EXTRACT(TIMEZONE_H    FROM DATE '2020-08-11');
+    //ERROR:  unit "timezone_h" not supported for type date
+
     @Test(expected = RuntimeException.class)
     // Calcite does not seem to know about DATE_TRUNC
```
Comment (lalithsuresh) [inline SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/postgres/PostgresDateTests.java:1448]: Can we add our own DATE_TRUNC for this?

File: SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/postgres/PostgresDateTests.java [hunk 2]
```diff
@@ -514,17 +1437,114 @@ public void testDoy() {
...
+    //
+    //select 'infinity'::date = '+infinity'::date as t;
+    // t
+    //---
+    // t
+
     @Test(expected = RuntimeException.class)
     // Calcite does not seem to support infinity dates
     public void testInfinity() {
         String query = "SELECT DATE 'infinity'";
```
Comment (lalithsuresh) [inline SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/postgres/PostgresDateTests.java:1533]: Anything we can do about this?

**Review by lalithsuresh — APPROVED** [review]

File: SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/postgres/PostgresDateTests.java [hunk 3]
```diff
@@ -355,21 +1270,32 @@ public void testDecade2() {
...
+    //ERROR:  unit "minute" not supported for type date
+    //SELECT EXTRACT(HOUR          FROM DATE '2020-08-11');
+    //ERROR:  unit "hour" not supported for type date
+
+    @Test
     public void testMillseconds() {
-        String query = "SELECT EXTRACT(MILLISECONDS  FROM DATE '2020-08-11')";
-        this.testQuery(query, new DBSPZSetLiteral(TypeCompiler.makeZSet(new DBSPTypeTuple())), true);
+        String query = "SELECT EXTRACT(MILLISECOND  FROM DATE '2020-08-11')";
+        this.testQueryTwice(query, 0);
```
[the remark this answers is not in the document]
Comment (mbudiu-vmw) [reply SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/postgres/PostgresDateTests.java:1294]: The second time is without "S" at the end.

File: SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/postgres/PostgresDateTests.java (hunk 2 above)
Comment (mbudiu-vmw) [reply SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/postgres/PostgresDateTests.java:1533]: We can do many things, the main question is whether it's worth our time or there are more useful things to do. I think infinity times can wait.

File: SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/postgres/PostgresDateTests.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (mbudiu-vmw) [reply SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/postgres/PostgresDateTests.java:1294]: sorry, misunderstood the question. 
It runs the compiler with -O0 and without it.
With -O0 there are no optimizations, and in particular, no constant folding. 
This checks that runtime evaluation of constant values is the same as compile-time evaluation.

[the remark this answers is not in the document]
Comment (lalithsuresh) [reply SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/postgres/PostgresDateTests.java:1294]: Just call it testQuery() then? The fact that it tests multiple configurations is fine.

File: SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/postgres/PostgresDateTests.java (hunk 1 above)
Comment (lalithsuresh) [reply SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/postgres/PostgresDateTests.java:1448]: I've done this for DCM (extending the parser). The annoying part is the build system, but after that it's pretty straightforward. Create an issue and assign it to me for now, i'll get to it in the coming week or two.

File: SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/postgres/PostgresDateTests.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (mbudiu-vmw) [reply SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/postgres/PostgresDateTests.java:1294]: testQueryTwice calls testQuery. Twice.

File: SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/postgres/PostgresDateTests.java (hunk 1 above)
Comment (mbudiu-vmw) [reply SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/postgres/PostgresDateTests.java:1448]: The main question is whether we are ready to fork calcite. That's a big step.
This should really be a Calcite issue...

Comment (lalithsuresh) [reply SQL-compiler/src/test/java/org/dbsp/sqlCompiler/compiler/postgres/PostgresDateTests.java:1449]: It's not a fork, that's the cool part. We basically create a folder like this where we add a few extra definitions: https://github.com/vmware/declarative-cluster-management/tree/master/dcm/src/main/codegen/includes

**Pull request merged**